Content.Server/Atmos/EntitySystems/DeltaPressureSystem.cs

```diff
@@ -0,0 +1,283 @@
+    }
+
+    /// <summary>
+    /// Attempts to remove an entity from the list of entities to do comparison and damage work on in AtmosphereSystem.
+    /// </summary>
+    /// <param name="ent">The entity to remove.</param>
+    /// <param name="grid">An EntityUid of the grid to remove this device from.</param>
+    /// <returns>True if the device was removed, false if the device could not be removed
+    /// from the provided grid EntityUid or was not in the processing list.</returns>
+    [PublicAPI]
+    public bool TryRemoveFromList(Entity<DeltaPressureComponent> ent, EntityUid grid)
+    {
+        return _atmosphereSystem.TryRemoveDeltaPressureEntity(grid, ent);
+    }
+
+    /// <summary>
+    /// Does damage to an entity depending on the pressure experienced by it, based on the
+    /// entity's <see cref="DeltaPressureComponent"/>.
+    /// </summary>
+    /// <param name="ent">The entity to apply damage to.</param>
+    /// <param name="pressure">The absolute pressure being exerted on the entity.</param>
+    /// <param name="deltaPressure">The delta pressure being exerted on the entity.</param>
+    [PublicAPI]
+    public void PerformDamage(Entity<DeltaPressureComponent> ent, float pressure, float deltaPressure)
+    {
+        var aboveMinPressure = pressure > ent.Comp.MinPressure;
+        var aboveMinDeltaPressure = deltaPressure > ent.Comp.MinPressureDelta;
+        if (!aboveMinPressure && !aboveMinDeltaPressure)
+        {
+            ent.Comp.IsTakingDamage = false;
+            return;
+        }
+
+        // shitcode
+        var baseDamage = ent.Comp.BaseDamage;
+        var appliedDamage = ent.Comp.BaseDamage;
+        if (aboveMinPressure)
+        {
+            appliedDamage = MutateDamage(ent, baseDamage, pressure - ent.Comp.MinPressure);
+        }
+        if (aboveMinDeltaPressure)
+        {
+            if (ent.Comp.StackDamage)
+            {
+                appliedDamage += MutateDamage(ent, appliedDamage, deltaPressure - ent.Comp.MinPressureDelta);
+            }
+            else
+            {
+                appliedDamage = MutateDamage(ent, baseDamage, deltaPressure - ent.Comp.MinPressureDelta);
+            }
+        }
+
+        _damage.TryChangeDamage(ent, appliedDamage, ignoreResistances: true, interruptsDoAfters: false);
+        ent.Comp.IsTakingDamage = true;
+    }
+
+    /// <summary>
+    /// Mutates the damage dealt by a DamageSpecifier based on values on an entity with a DeltaPressureComponent.
+    /// </summary>
+    /// <param name="ent">The entity to base the manipulations off of (pull scaling type)</param>
+    /// <param name="damage">The damage specifier to mutate.</param>
+    /// <param name="pressure">The pressure being exerted on the entity.</param>
+    /// <returns>The mutated DamageSpecifier.</returns>
+    [PublicAPI]
+    public static DamageSpecifier MutateDamage(Entity<DeltaPressureComponent> ent, DamageSpecifier damage, float pressure)
```

Comment:
I really don't think this should be mutating stuff like this. I don't see any reason why it can't be pure and just create a new damage specifier if needed.